test/parallel/test-heapdump-fs-promise.js

```diff
@@ -0,0 +1,16 @@
+// Flags: --expose-internals
+'use strict';
+require('../common');
+const { validateSnapshotNodes } = require('../common/heap');
+const fs = require('fs').promises;
+
+validateSnapshotNodes('FSREQPROMISE', []);
+fs.stat(__filename);
+validateSnapshotNodes('FSREQPROMISE', [
+  {
+    edges: [
+      { name: 'FSReqPromise' },
+      { name: 'Float64Array' }  // Stat array
```

Comment:
If I am reading the code correctly...this means that there will be edges named with class names? Aren't those supposed to be something like property names? (e.g. something like `fs_stats_field_array` here)

Reply:
As you edited above, we don’t currently have any ways to name the edges as embedders :/

Reply:
> but until it's sorted out it's still worth landing this with indexed edges IMO

@joyeecheung Can you explain, what does “indexed edges” mean here? i.e. how is it different from what this PR currently does? Or am I missing something? :smile:

Reply:
@addaleax The current EmbedderGraph API will create edges with type `kElement` and with indexes that are automatically incremented (based on the count of edges of the from node). The internal V8 edges, for example, the ones pointing to `shared function info` or `context` nodes from JS functions, have type `kInternal` and corresponding string names. That's why those references show up in the DevTools as gray and will have proper names displayed (otherwise you may see a function has a reference named with a number to a `shared function info` node, instead of a reference named `shared`).

Reply:
@joyeecheung Okay, I understand. So you’re okay with the current approach and we can update to a better API once we have that in V8?

Reply:
@addaleax Yeah, if I understand correctly the snapshots generated with the current approach will show up with number indexed references from the embedder nodes to other nodes, which would look a bit confusing but still it's better than no references tracked in the snapshots at all.

I opened https://chromium-review.googlesource.com/c/v8/v8/+/926362 BTW

Reply:
@addaleax Ooops, wrong link: https://chromium-review.googlesource.com/c/v8/v8/+/1133299

Reply:
<img width="335" alt="screen shot 2018-07-12 at 12 36 22 am" src="https://user-images.githubusercontent.com/4299420/42586898-b292e482-856b-11e8-91ef-b4c3a0a9075a.png">

By the way this is what I meant by "indexed edges", which are also created by the deprecated `RetainedObjectInfo`

Reply:
Okay, yeah, makes sense – not the end of the world but would definitely be very nice to have.